wallet/src/de/schildbach/wallet/ui/ExportTransactionHistoryDialogBuilder.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2013-2015 the original author or authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.ui;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.core.content.FileProvider;
+
+import com.google.common.base.Charsets;
+
+import org.dash.wallet.common.ui.BaseAlertDialogBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+
+import javax.annotation.Nullable;
+
+import de.schildbach.wallet.Constants;
+import de.schildbach.wallet.WalletApplication;
+import de.schildbach.wallet.util.CrashReporter;
+import de.schildbach.wallet.util.WalletUtils;
+import de.schildbach.wallet_test.R;
+import kotlin.Unit;
+
+/**
+ * @author Eric Britten
+ */
+public abstract class ExportTransactionHistoryDialogBuilder extends BaseAlertDialogBuilder {
+    private final Activity context;
+
+    private final EditText viewDescription;
```

Comment:
This is copied from `CrashReporterDialogBuilder`.  Some things are removed that are not needed.

Future designs will eliminate the need for this dialog, but the code that is related to sharing an attachment with the transaction history will still be needed. 

I suggest that we can update, refactor or rewrite the code related to exporting files in the story will be adding the UI and associated features.